lib/instrumentation/http-shared.js

```diff
@@ -143,7 +143,12 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
       ins.bindEmitter(req)
 
       span.name = req.method + ' ' + req._headers.host + url.parse(req.path).pathname
-      req.on('response', onresponse)
+
+      const emit = req.emit
+      req.emit = function (type, res) {
+        if (type === 'response') onresponse(res)
+        return emit.apply(req, arguments)
+      }
```

Comment:
Is there some reason you didn't use shimmer here?

Reply:
I was concerned that maybe the emit function had already been shimmed and therefore the 2nd call would be ignored. But maybe that's not the case 🤔

Reply:
Hmm...okay. It'd be nice if there was a better way to prevent multiple patching that didn't prevent _intentional_ multiple patching? 🤔

Reply:
I'll update it to use shimmer

Reply:
Oh wait, I actually remember another bigger reason now why I didn't use shimmer: This happens for every outgoing http request and I wanted to reduce the overhead as much as possible and the `shimmer.wrap` function has a slightly larger overhead. But it's probably not that big of a difference, so maybe I should just use shimmer?

Reply:
Can we find a way to apply this patch on the prototype instead? I'd prefer we avoid building a new closure for every call. 🤔

Reply:
Actually, if we're planning a rewrite anyway, it's probably fine to just land this as-is to fix the memory leak issue and just take a note somewhere to look at moving the patch to the prototype in the rewrite.

Reply:
👍 I'll add a TODO comment